src/web/components/Callout/Form.tsx

```diff
@@ -0,0 +1,180 @@
+import React, { useState } from 'react';
+import { css } from 'emotion';
+
+import { Link } from '@guardian/src-link';
+import { Button } from '@guardian/src-button';
+import { text } from '@guardian/src-foundations/palette';
+import { textSans } from '@guardian/src-foundations/typography';
+
+import { CheckboxList } from './CheckboxList';
+import { FileUpload } from './FileUpload';
+import { RadioList } from './RadioList';
+import { Select } from './Select';
+import { TextArea } from './TextArea';
+import { TextInput } from './TextInput';
+
+const formStyles = css`
+    display: flex;
+    flex-direction: column;
+    justify-content: space-between;
+    padding-left: 10px;
+    padding-right: 10px;
+`;
+
+const footerPaddingStyles = css`
+    display: flex;
+    flex-direction: row;
+    justify-content: space-between;
+`;
+
+const errorMessagesStyles = css`
+    padding-bottom: 10px;
+    color: ${text.error};
+    ${textSans.medium({ fontWeight: 'bold' })};
+`;
+
+type formData = { [key in string]: any };
+
+type FormFieldProp = {
+    formField: CampaignFieldType;
+    formData: formData;
+    setFormData: React.Dispatch<React.SetStateAction<formData>>;
+};
+
+const FormField = ({ formField, formData, setFormData }: FormFieldProp) => {
+    switch (formField.type) {
+        case 'textarea':
+            return (
+                <>
+                    <TextArea
+                        formField={formField}
+                        formData={formData}
+                        setFormData={setFormData}
+                    />
+                    <hr />
+                </>
+            );
+        case 'file':
+            return (
+                <>
+                    <FileUpload
+                        formField={formField}
+                        formData={formData}
+                        setFormData={setFormData}
+                    />
+                    <hr />
+                </>
+            );
+        case 'select':
+            return (
+                <>
+                    <Select
+                        formField={formField}
+                        formData={formData}
+                        setFormData={setFormData}
+                    />
+                    <hr />
+                </>
+            );
+        case 'radio':
+            return (
+                <>
+                    <RadioList
+                        formField={formField}
+                        formData={formData}
+                        setFormData={setFormData}
+                    />
+                    <hr />
+                </>
+            );
+        case 'checkbox':
+            return (
+                <>
+                    <CheckboxList
+                        formField={formField}
+                        formData={formData}
+                        setFormData={setFormData}
+                    />
+                    <hr />
+                </>
+            );
```

Comment:
This is still confusing. If we're returning an array of things then why are we passing it `formField`? My brain jars when it sees this. This might seem like nitpicking but incorrect use of plurals is one of the biggest sources of confusion and can cause hard to find bugs.

I'm actually confused as to what this _should_ be because I actually can't tell from the code what type of data this is, which is a flag that there are possible improvements to be had!

Reply:
In the past, when I've built forms with radios and checkboxes I've reserved the words 'radio' and 'checkbox' to only describe the actual circles or squares that a user can click on. Instead I create a 'MultiSelect' or 'Choices' or 'OptionsGroup' or something, which is the grouping of multiple choices. That component can then have a prop `multiple: boolean` which decides if radios or checkboxes are used.

Reply:
Finding the right level of abstraction is not always clear.

I see your point, I dnt like the names `CheckboxList` and `RadioList`.

From what I gather you would prefer to lift the logic of looping through options here? If so we would then run into the same issue of having unnecessary re-renders, because we loose the local state of `CheckboxList` and `RadioList`.

Reply:
> `MultiSelect`
that sounds like a much better approach! I will implement that. Thanks